plugins/auth-backend/src/providers/auth0/provider.ts

```diff
@@ -53,18 +54,38 @@ export type Auth0AuthProviderOptions = OAuthProviderOptions & {
   signInResolver?: SignInResolver<OAuthResult>;
   authHandler: AuthHandler<OAuthResult>;
   resolverContext: AuthResolverContext;
+  audience?: string;
 };
 
 export class Auth0AuthProvider implements OAuthHandlers {
   private readonly _strategy: Auth0Strategy;
   private readonly signInResolver?: SignInResolver<OAuthResult>;
   private readonly authHandler: AuthHandler<OAuthResult>;
   private readonly resolverContext: AuthResolverContext;
+  private readonly audience?: string;
+
+  /**
+   * Due to passport-auth0 forcing options.state = true,
```

Comment:
At a glance it seems to me like we could instead pass in `state: false` in the strategy options, avoiding the usage of the session. Does that work?

Reply:
no because the `passport-auth0` implementation then bans the use of `openid` scope at this point https://github.com/auth0/passport-auth0/blob/master/lib/index.js#L94. Hence I'm blatantly stealing from the backstage okta implementation here https://github.com/backstage/backstage/blob/master/plugins/auth-backend/src/providers/okta/provider.ts#L79 and https://github.com/backstage/backstage/blob/master/plugins/auth-backend/src/providers/okta/provider.ts#L102. This is because the session store never gets saved for some reason in the default way. 

As they say "good artists borrow, great artists steal" :)

Reply:
Oh dear, alright so they forcefully block OpenID auth without a session because there's no state verification, but we do that verification ourselves outside the strategy so that's all good. I do wonder why OpenID specifically is blocked since OAuth in general is less secure without state verification. Oh well, workaround it is then x)